src/cli.rs

```diff
@@ -96,6 +96,10 @@ pub struct Cli {
     /// Optional key-value pairs to be included in the request.
     #[structopt(name = "REQUEST_ITEM")]
     pub request_items: Vec<RequestItem>,
+
+    /// Skip the host's SSL certificate verification
+    #[structopt(long)]
+    pub verify: Option<VerifyHttps>,
```

Comment:
I was working on adding these exact features, and I see you've chosen a slightly different solution. I thought it might make more sense to have the `--verify` flag have a default value (which is `yes`), instead of making it optional.

To achieve this, I used the `default_value` macro _parameter_ (dunno if that's what they're called, I'm a bit of a Rust noobie), and had an implementation for the `Default` trait, like so:
```rust
impl Default for VerifyHttps {
  fn default() -> Self {
    VerifyHttps::Yes
  }
}
```

I feel this better resembles what httpie does, and it rids the codebase of an `Option` you'll have to deal with.

Reply:
And seeing how I had the part ready that disables verification, here's the diff I had, which you could use in this PR if you want:

```diff
diff --git a/src/main.rs b/src/main.rs
index cc896ef..91d4534 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -18,7 +18,7 @@ mod utils;
 use anyhow::{anyhow, Result};
 use auth::Auth;
 use buffer::Buffer;
-use cli::{AuthType, Cli, Method, Pretty, Print, RequestItem, Theme};
+use cli::{AuthType, Cli, Method, Pretty, Print, RequestItem, Theme, VerifyHttps};
 use download::{download_file, get_file_size};
 use printer::Printer;
 use request_items::{Body, RequestItems};
@@ -68,8 +68,15 @@ async fn main() -> Result<()> {
         true => Policy::limited(args.max_redirects.unwrap_or(10)),
         false => Policy::none(),
     };
+    let disable_certificate_verification = match args.verify {
+        VerifyHttps::No => true,
+        _ => false,
+    };
 
-    let client = Client::builder().redirect(redirect).build()?;
+    let client = Client::builder()
+        .redirect(redirect)
+        .danger_accept_invalid_certs(disable_certificate_verification)
+        .build()?;
     let request = {
         let mut request_builder = client
             .request(method, url.0)
```

**NB:** I haven't yet finished much work on implementing the custom CA bundle functionality, so this would still have to be done.

Reply:
Thanks a lot! I've updated the code.